src/collector.js

```diff
@@ -76,6 +77,10 @@ export default class Collector {
     return false;
   }
 
+  recordScannedFile(filename) {
+    this.scannedFiles.push(filename);
```

Comment:
I agree, I'll change it to record a structure similar to that. That'll also resolve the dupes problem and make the method more worthwile to keep around.